guarddog/analyzer/metadata/typosquatting.py

```diff
@@ -253,6 +252,9 @@ def detect(self, package_info) -> tuple[bool, Optional[str]]:
             list[str]: names of packages that <package_name> could be
             typosquatting from
         """
+        if ecosystem == "npm":
+            return False, "This rule does not work for npm yet"
```

Comment:
will that print an error message by default when running on an NPM package? I think it's a symptom that we need to make heuristics specific to a single ecosystem instead of having a bunch of "if/else" in their definition

Reply:
Yeah I think that would be great. Ideally with 1 `Package` type per ecosystem?

Reply:
If that works for you @christophetd let's do it in another PR (that has to figure injection of heuristics)? I'll start to work on it right away

Reply:
Ok, I have an idea. Adding in to this PR